flink-core/src/main/java/org/apache/flink/core/plugin/DirectoryBasedPluginFinder.java

```diff
@@ -78,19 +78,17 @@ public DirectoryBasedPluginDescriptorsFactory(Path pluginsRootDir) {
 
 	private Optional<PluginDescriptor> createPluginDescriptorForSubDirectory(Path subDirectory) {
 		Optional<URL[]> jarURLsFromDirectory = createJarURLsFromDirectory(subDirectory);
-		if (jarURLsFromDirectory.isPresent()) {
-			URL[] urls = jarURLsFromDirectory.get();
-			// we sort the urls for the benefit of having a stable/reproducible order of jars.
-			Arrays.sort(urls, Comparator.comparing(URL::toString));
-			//TODO: This class could be extended to parse exclude-pattern from a optional text files in the plugin directories.
-			return Optional.of(
-				new PluginDescriptor(
-					subDirectory.getFileName().toString(),
-					urls,
-					new String[0]));
-		} else {
-			return Optional.empty();
-		}
+		return jarURLsFromDirectory
+			.map((URL[] urls) -> {
+				Arrays.sort(urls, Comparator.comparing(URL::toString));
+				//TODO: This class could be extended to parse exclude-pattern from a optional text files in the plugin directories.
+				return Optional.of(
+					new PluginDescriptor(
+						subDirectory.getFileName().toString(),
+						urls,
+						new String[0]));
+			})
+			.orElse(Optional.empty());
```

Comment:
I think you could drop `.orElse(...)` line.

Reply:
I think you saw some outdated version, this was already removed :)